packages/protocol/contracts/layer1/shasta/impl/ForcedInclusionStore.sol

```diff
@@ -94,45 +94,62 @@ contract ForcedInclusionStore2 is EssentialContract, IForcedInclusionStore {
     }
 
     /// @inheritdoc IForcedInclusionStore
-    /// @dev Only the inbox contract can call it since we don't do any validation here
-    function consumeOldestForcedInclusion(address _feeRecipient)
+    function consumeForcedInclusions(
+        address _feeRecipient,
+        uint256 _count
+    )
         external
         onlyFrom(inbox)
         nonReentrant
-        returns (ForcedInclusion memory inclusion_)
+        returns (ForcedInclusion[] memory inclusions_)
     {
-        // we only need to check the first one, since it will be the oldest.
-        ForcedInclusion storage inclusion = queue[head];
-        require(inclusion.blobSlice.timestamp != 0, NoForcedInclusionFound());
+        // Early exit if no inclusions requested or queue is empty
+        if (_count == 0 || head == tail) {
+            return new ForcedInclusion[](0);
+        }
 
-        inclusion_ = inclusion;
+        // Calculate actual number to process (min of requested and available)
+        uint256 available = tail - head;
+        uint256 toProcess = _count > available ? available : _count;
 
-        lastProcessedAt = uint64(block.timestamp);
+        inclusions_ = new ForcedInclusion[](toProcess);
+        uint256 totalFees;
 
         unchecked {
-            delete queue[head++]; // delete element at head AND THEN increment head
-            _feeRecipient.sendEtherAndVerify(inclusion_.feeInGwei * 1 gwei);
+            for (uint256 i; i < toProcess; ++i) {
+                ForcedInclusion storage inclusion = queue[head + i];
+                inclusions_[i] = inclusion;
```

Comment:
how about `inclusions_[i] = queue[head + i];` and we use `inclusions_[i] ` as it is already in memory